### PR TITLE
Add more tests for recently-fixed dynamic bugs

### PR DESCRIPTION
**Pull request by khyperia**
More tests for issues #9945 and #10463 as discussed in PRs #11219 and #11256 as well as in-person.

Ping @dotnet/roslyn-compiler for review.

### CODE REVIEW

File: src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTests.cs [hunk 1]
```diff
@@ -14902,5 +14902,414 @@ .maxstack  10
...
+        }
+        catch (Microsoft.CSharp.RuntimeBinder.RuntimeBinderException)
+        {
+            Console.Write(""ex caught"");
+        }
+    }
+}
+";
+
+            // TODO: Figure out if we want this error or not.
```
Comment (agocke) [inline src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTests.cs]: It looks like this is a breaking change vs native -- native used to let this pass through.

We also have a policy on no bare TODOs: all TODOs must have issues filed.

**Comment by AlekseyTs** [comment]
@khyperia We need to expedite the decision on whether we want to keep the break or not.

**Comment by gafter** [comment]
@AlekseyTs I'll close this issue with the compat team tomorrow. So far we have tentative approval and no objections.

**Comment by khyperia** [comment]
I've updated this assuming we decided to keep the breaking change, but I believe a final decision hasn't been reached yet (still tentative, as Neal said). Avoid merging until that's final, should be later today (assuming the decision doesn't change).

Rebased to grab `Compiler Breaking Changes.md`, but the only changes from previously are the most recent commit.

**Comment by gafter** [comment]
@khyperia We have approval from the compat council.
LGTM :shipit: :+1:

File: docs/compilers/CSharp/Compiler Breaking Changes.md [hunk 2]
```diff
@@ -6,3 +6,6 @@ Each entry should include a short description of the break, followed by either a
 1. In some cases, due to a bug in the native compiler, programs with pointers to structs with one or more
    type parameters compiled without error. All such programs should now produce errors in Roslyn. See
    [#5712](https://github.com/dotnet/roslyn/issues/5712) for examples and details.
+2. When calling a method group with only instance methods in a static context with dynamic arguments, the
```
Comment (agocke) [inline docs/compilers/CSharp/Compiler Breaking Changes.md]: As it says at the top, please link the bug.

**Comment by jaredpar** [comment]
:+1:

**Pull request merged**